rotkehlchen/db/dbhandler.py

```diff
@@ -219,9 +220,13 @@ def __init__(
         # Lock to make sure that 2 callers of get_or_create_evm_token do not go in at the same time
         self.get_or_create_evm_token_lock = Semaphore()
         self.password = password
-        self._connect()
-        self._check_unfinished_upgrades(resume_from_backup=resume_from_backup)
-        self._run_actions_after_first_connection()
+        try:
+            self._connect()
+            self._check_unfinished_upgrades(resume_from_backup=resume_from_backup)
+            self._run_actions_after_first_connection()
+        except (SystemPermissionError, AuthenticationError, RotkehlchenPermissionError, DBUpgradeError, DBSchemaError):  # noqa: E501
+            self.logout()
+            raise
```

Comment:
I guess this errors existed before right? Maybe add a comment for why you need the logout since we might forget in the future.

Question: in windows if there was an error here what happened? Did the app work?

Reply:
this is called when we create a new user or log in an existing user. All these error are handled in `user_login` and the response is sent back accordingly

Reply:
So if it's all happening there why do you need this here?

Can't you just add it in `reset_after_failed_account_creation_or_login()`:

```python
        try:
            self.rotkehlchen.unlock_user(
                user=name,
                password=password,
                create_new=False,
                sync_approval=sync_approval,
                premium_credentials=None,
                resume_from_backup=resume_from_backup,
            )
        # We are not catching PremiumAuthenticationError here since it should not bubble
        # up until here. Even with non valid keys in the DB login should work fine
        except AuthenticationError as e:
            self.rotkehlchen.reset_after_failed_account_creation_or_login()
            return wrap_in_fail_result(
                message=str(e),
                status_code=HTTPStatus.UNAUTHORIZED,
            )
        except RotkehlchenPermissionError as e:
            self.rotkehlchen.reset_after_failed_account_creation_or_login()
            return {
                'result': e.message_payload,
                'message': e.error_message,
                'status_code': HTTPStatus.MULTIPLE_CHOICES,
            }
        except (DBUpgradeError, SystemPermissionError, DBSchemaError) as e:
            self.rotkehlchen.reset_after_failed_account_creation_or_login()
            return wrap_in_fail_result(
                message=str(e),
                status_code=HTTPStatus.CONFLICT,
            )
        except sqlcipher.OperationalError as e:  # pylint: disable=no-member
            self.rotkehlchen.reset_after_failed_account_creation_or_login()
            return wrap_in_fail_result(
                message=f'Unexpected database error: {e!s}',
                status_code=HTTP_STATUS_INTERNAL_DB_ERROR,  # type: ignore  # Is a custom status code, not a member of HTTPStatus
            )
```

Reply:
so the initial reason was because of the tests. In many unit tests we directly initialize the db without the rest API, so there the fallback cleanup was not being done. But also, I think cleanup related to the db happening in the same class would better because these similar logic are already present here and this would later make it easy when the rest API gets served from a seperate process